h2/stream.py

```diff
@@ -918,11 +930,12 @@ def receive_headers(self, headers, end_stream, header_encoding):
             if not end_stream:
                 raise ProtocolError("Trailers must have END_STREAM set")
 
-        conn_state = HeaderValidationFlags(
-            is_client=self.state_machine.client,
-            is_trailer=isinstance(events[0], TrailersReceived)
-        )
-        headers = validate_headers(headers, conn_state)
+        if self.config.validate_inbound_headers:
+            hdr_validation_flags = HeaderValidationFlags(
+                is_client=self.state_machine.client,
+                is_trailer=isinstance(events[0], TrailersReceived)
+            )
```

Comment:
This can also use `_build_hdr_validation_flags` if we slightly broaden that method.